game-core/src/main/java/games/strategy/engine/data/RouteFinder.java

```diff
@@ -16,27 +17,29 @@
 import java.util.Set;
 import java.util.function.Function;
 import java.util.function.Predicate;
+import javax.annotation.Nullable;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
 
+@RequiredArgsConstructor(access = AccessLevel.PACKAGE, onConstructor_ = @VisibleForTesting)
 class RouteFinder {
 
+  private final MoveValidator moveValidator;
```

Comment:
I'm not entirely convinced this field is really necessary.
The `GameMap` object is still there, so why not create this object on the fly inside of `getNeighborsValidatingCanals`?

Reply:
Creating an object on the fly is a violation of IOC. It would be equivalent to a static method call, which would be an example of static coupling. The intent was to get a mock in there so that this class could be tested, 'map' is already mocked and I pretty much assumed the move validator path was going to need the same.

In the short term, raising the dependency from a method where there is in effect a static dependency to a constructor injection can help us decouple this class from 'gameData' (which in turn comes from gameMap). This class is already very odd as the predicate member probably should *not* be a constructor arg..